### PR TITLE
example of a rainy day/night zone

### PR DESCRIPTION
**Pull request by ZappoMan**
This is a self contained Zone that implements a rainy day/night. It can be used as a zone, which includes a shader or the rain sound and thunder/lightening light/sound effect can be added to any other entity in a domain.

To use:
- Option 1 - import RainyDayNightZone.json into your domain - this will give you everything
- Option 2 - add lighteningEntity.js as a script to any entity - this will give you the lightening/thunder and rain sounds.
- Opton 3 - if you just want the rain effect on your skydome, without the thunder and lightening, you can use the rainyDayNightSkybox.fs shader as a custom shader for a zone

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a752f55deb4b1ebb5f70466e59f4f2cc1620852f available at: https://gist.github.com/d73970b1e94354a1d35f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.dmg

**Comment by OmegaHeron** [comment]
Very cool!  Loaded json via import and tried the components as well.  Fun!  Matches my RL part of world well - it's been too dry in HF for a Floridian.  :P

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: a752f55deb4b1ebb5f70466e59f4f2cc1620852f available at: https://gist.github.com/d73970b1e94354a1d35f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.dmg

File: examples/entityScripts/lighteningEntity.js [hunk 1]
```diff
@@ -0,0 +1,339 @@
+//
+//  lighteningEntity.js
```
Comment (jherico) [inline examples/entityScripts/lighteningEntity.js]: *lightning*

File: examples/shaders/rainyDayNightSkybox.fs [hunk 2]
```diff
@@ -0,0 +1,126 @@
+// Created by inigo quilez - iq/2013
+// Turbulence and Day/Night cycle added by Michael Olson - OMGparticles/2015
+// rain effect adapted from Rainy London by David Hoskins. - https://www.shadertoy.com/view/XdSGDc
+// License Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License.
+#line 5
```
Comment (jherico) [inline examples/shaders/rainyDayNightSkybox.fs]: the line pragma refers to the *next* line of the shader.  This should be `#line 6`

File: examples/shaders/rainyDayNightSkybox.fs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    col += 0.4 * vSunRimColor * pow( sun,    4.0 );
+    col += 1.0 * vSunColor    * pow( sun, 2000.0 );
+    
+    // stars
+    float fStarContrib = clamp((fSunHeight - fDayHeight) * (-fInverseHL), 0.0, 1.0);
+    
+    vec3 vStarDir = rd;
+                              
+    col = mix(col, stars3(vStarDir), fStarContrib);
+    col += stars3(vStarDir) * fStarContrib;
```
Comment (jherico) [inline examples/shaders/rainyDayNightSkybox.fs]: This makes no sense and may be the cause of the weird color shimmer during the day night transition.  I think this line is superfluous / harmful.

File: examples/shaders/rainyDayNightSkybox.fs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        vec2 st =  f * (q * vec2(2.0, .05) + vec2(-iGlobalTime*.01 + q.y * .05, iGlobalTime * 0.12));
+        f = texture2D(iChannel2, st * .5, -59.0).x;
+        f += texture2D(iChannel2, st*.284, -99.0).y;
+        f = clamp(pow(abs(f)*.5, 29.0) * 140.0, 0.00, q.y*.4+.05);
+        vec3 bri = vec3(rainBrightness);
+        col += bri*f;
+		dis += 3.5;
+	}
+
+    return vec4( col, 1.0 );
```
Comment (jherico) [inline examples/shaders/rainyDayNightSkybox.fs]: col should be clamped to between 0 and 1 before being returned.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d56a0d05167ac09ddc5fe821dd4e83e63b86a0e available at: https://gist.github.com/d73970b1e94354a1d35f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2b4d206eca12a909170c1ff0136041689dbe854f available at: https://gist.github.com/d73970b1e94354a1d35f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7242.dmg

**Pull request merged**